security/pkg/nodeagent/caclient/providers/google-cas/client.go

```diff
@@ -0,0 +1,157 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package caclient
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	privateca "cloud.google.com/go/security/privateca/apiv1"
+	"google.golang.org/api/option"
+	privatecapb "google.golang.org/genproto/googleapis/cloud/security/privateca/v1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/test/bufconn"
+	"google.golang.org/protobuf/types/known/durationpb"
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	"istio.io/istio/pkg/security"
+	"istio.io/istio/security/pkg/nodeagent/caclient"
+	"istio.io/istio/security/pkg/nodeagent/caclient/providers/google-cas/mock"
+	"istio.io/pkg/log"
+)
+
+var googleCAClientLog = log.RegisterScope("googlecas", "Google CAS client debugging", 0)
+
+type GoogleCASClient struct {
+	caSigner string
+	caClient *privateca.CertificateAuthorityClient
+	provider *caclient.TokenProvider
+}
+
+// NewGoogleCASClient create a CA client for Google CAS.
+func NewGoogleCASClient(capool string, provider *caclient.TokenProvider, lis *bufconn.Listener) (security.Client, error) {
+	var err error
+	caClient := &GoogleCASClient{caSigner: capool, provider: provider}
+	ctx := context.Background()
+
+	if lis == nil {
+		caClient.caClient, err = privateca.NewCertificateAuthorityClient(ctx,
+			option.WithGRPCDialOption(grpc.WithPerRPCCredentials(provider)))
+	} else {
+		caClient.caClient, err = privateca.NewCertificateAuthorityClient(ctx,
+			option.WithoutAuthentication(),
+			option.WithGRPCDialOption(grpc.WithContextDialer(mock.ContextDialerCreate(lis))),
+			option.WithGRPCDialOption(grpc.WithInsecure()))
+	}
+	if err != nil {
+		googleCAClientLog.Errorf("unable to initialize google cas caclient: %v", err)
+		return nil, err
+	}
+	return caClient, nil
+}
+
+func (r *GoogleCASClient) createCertReq(name string, csrPEM []byte, lifetime time.Duration) *privatecapb.CreateCertificateRequest {
+	subjectConfig := &privatecapb.CertificateConfig_SubjectConfig{
+		// Empty Subject
+		Subject: &privatecapb.Subject{},
+	}
+	keyUsage := &privatecapb.KeyUsage{
```

Comment:
nit: can this be inlined below for clarity?

(and same for all other sections of the cert request -- having them inline would make it easier to figure out what gets included, since each variable is only used once)